deploy/lib/server_config.rb

```diff
@@ -1362,6 +1379,41 @@ def settings
 
 private
 
+  def filter_to_newer_files(files, options)
+    logger.info "Filtering to files which are newer locally than on the server"
+
+    if @server_version < 6
+      raise ExitException.new("Can only filter files on MarkLogic 6 and later")
+    end
+
+    uris = files.map { |f| xcc.build_target_uri(f, options) }
+    stamps_db = get_db_timestamps(uris)
+    stamps_local = files.map { |file_uri| File.mtime(file_uri).getgm.iso8601(5).tr('Z','') }
```

Comment:
The local timestamps are accounting for timezone, but get_db_timestamps() doesn't. In my case, that meant my local timestamps were 5 hours ahead, so all files needed to be loaded. After removing the `.getgm` and the `.tr(...)` from the line above, Ruby compared timestamps that looked like:

- local: 2015-12-17T13:13:30.00000-05:00
- in db: 2015-12-17T13:10:00.229146

While they look slightly different, they are in the same timezone.

To make this work for remote servers, I think you'll need to call [fn:implicit_timezone()](http://docs.marklogic.com/fn:implicit-timezone) on the MarkLogic server, then use that value with [Time.getlocal(utc_offset)](http://ruby-doc.org/core-2.2.0/Time.html#method-i-getlocal) for the local times.